src/r2mm/downloading/BetterThunderstoreDownloader.ts

```diff
@@ -1,117 +1,36 @@
-        deps.sort((a, b) => {
-            if (a.getVersion().getDependencies().find(value => value.startsWith(b.getMod().getFullName() + "-"))) {
-                return 1;
-            } else {
-                return -1;
-            }
-        })
-    }
-
-    public async downloadLatestOfAll(modsWithUpdates: ThunderstoreCombo[], ignoreCache: boolean,
-                                      callback: (progress: number, modName: string, status: number, err: R2Error | null) => void,
-                                      completedCallback: (modList: ThunderstoreCombo[]) => void) {
-
-        const dependencies: ThunderstoreCombo[] = [...modsWithUpdates];
-
-        for (const mod of modsWithUpdates) {
-            await this.buildDependencySet(mod.getVersion(), dependencies, DependencySetBuilderMode.USE_LATEST_VERSION);
-        }
-
-        this.sortDependencyOrder(dependencies);
-
-        let downloadCount = 0;
-        const downloadableDependencySize = this.calculateInitialDownloadSize(dependencies);
-
-        await this.queueDownloadDependencies(dependencies.entries(), ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-            if (status === StatusEnum.FAILURE) {
-                callback(0, modName, status, err);
-            } else if (status === StatusEnum.PENDING) {
-                callback(generateProgressPercentage(progress, downloadCount, downloadableDependencySize + 1), modName, status, err);
-            } else if (status === StatusEnum.SUCCESS) {
-                callback(generateProgressPercentage(progress, downloadCount, downloadableDependencySize + 1), modName, StatusEnum.PENDING, err);
-                downloadCount += 1;
-                if (downloadCount >= dependencies.length) {
-                    callback(100, modName, StatusEnum.PENDING, err);
-                    completedCallback([...dependencies]);
-                }
-            }
-        });
-    }
-
-    public async download(
-        profile: ImmutableProfile,
-        combo: ThunderstoreCombo,
+    public async downloadCombos(
+        combos: ThunderstoreCombo[],
         ignoreCache: boolean,
         totalProgressCallback: (progress: number, modName: string, status: number, err: R2Error | null) => void
     ): Promise<ThunderstoreCombo[]> {
-
-        const modList = await ProfileModList.getModList(profile);
-        if (modList instanceof R2Error) {
-            totalProgressCallback(0, combo.getMod().getName(), StatusEnum.FAILURE, modList);
-            throw modList;
-        }
-
-        const dependencies = await this.getDependenciesWithCorrectVersions(combo, modList);
-        const allModsToDownload = [combo, ...dependencies];
-
-        let modInProgressName = combo.getMod().getName();
+        let modInProgressName = combos[0].getMod().getName();
         let downloadCount = 0;
-        // Mark the mod 80% processed when the download completes, save the remaining 20% for extracting.
```

Comment:
Was this removed intentionally? I kinda like it as it explains the reason for magic number `0.8` below.

Reply:
I agree, I added this back 👍